PyInstaller/hooks/hook-PyQt5.QtQml.py

```diff
@@ -36,3 +38,481 @@
+        with open(self.spec_file, mode='rb') as spec:
+
+            data = spec.read()
+
+        # find the line that contains the analysis variable
+        analysis = re.findall('a = .*?.*?.*?,', str(data))
+
+        # strip extraneous data
+        without_com = analysis[0].replace(',', '')
+        without_analy_stat = without_com.replace('a = Analysis(', '')
+        without_apos = without_analy_stat.replace("'", "")
+        a_right = without_apos[1:-1]
+
+        # the main python file
+        self.analysis_file = a_right
+        if os.path.exists(self.analysis_file):
+            path_splits = os.path.split(self.analysis_file)
+            # the folder the main py file is in
+            self.analysis_folder = path_splits[0]
+            return True
+        else:
+            return False
+
+    def read_analys_file(self):
+
+        # This is where we read the main python file
+        # to find the main qml file used.
+
+        with open(self.analysis_file, mode='rb') as analys:
+            for line in analys:
+                if re.findall('[ ]+[#]', str(line)):
+                    # skip it is a comment in the code
+                    continue
+                else:
+                    # find line that contains a qml file
+                    for search_query in self.search_words:
+                        query = '.' + search_query + '[(].*?.*?.*?[)]'
+                        hyp_file = re.findall(query, str(line))
+
+                        if hyp_file:
+                            # we will use the first file
+                            first = hyp_file[0].replace(search_query, '')
+                            second = first[2:-1]
+                            if self.is_raw_string(second):
+                                return True
+                            else:
+                                return False
+                            break
+
+                        else:
+                            continue
+
+    def is_raw_string(self, raw):
+
+        # Here we read the line to see if there is
+        # a genuine qml filepath.
+        if re.findall('([+]|,|[)])', raw) or not re.findall('(\'|")', raw):
+            # It is a variable
+            return False
+
+        else:
+            if re.findall('qrc:', raw):  # it is a qrc
+                return False
+            else:
+                if raw[0] == '"':
+                    self.main_qml = raw.replace('"', '')
+                    return True
+
+                elif raw[0] == "'":
+                    self.main_qml = raw.replace("'", "")
+                    return True
+                else:
+                    return False
+
+    def reconstruct_qml_path(self):
+
+        # This is where we verify if the
+        # qml filepath is a relative path
+        # to the application's folder
+        # if it's not, we simply do nothing.
+        # it can't be a full path.
+        if os.path.exists(self.main_qml):
+            return False
```

Comment:
No need to return False. Just reverse the test and ose only the `else` part.

Beside this:
Shall this test whether the path exists (as the code does) or whether the path is a relative path (as the comment says)?
If the later, the correct check would be ` if os.path.isabs(string)`